app/src/main/res/values/themes_authentication.xml

```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="AppTheme.Authentication">
+        <item name="android:colorBackground">#f8f8f8</item>
+        <item name="android:textColorPrimary">@color/graphite</item>
```

Comment:
I'm not sure how to proceed here. If there's a non-repeating color:
should I create a color palette and reference those colors here (as in textColorPrimary),
or should I directly add the hex color value here (as in colorBackground)?

let me know what you think :)

Reply:
Not sure how reputable this [project](https://github.com/futurice/android-best-practices) is, but, they do offer some [guidance](https://github.com/futurice/android-best-practices#colorsxml) on colours in particular with styles. I personally like their approach.

For single use colours, maybe we need to ask why we're only using this colour once in the app? This might help us mitigate a polluted `colors.xml` file like we have in wire-android. But, normally as a rule of thumb I try and avoid hardcoded values where I can outside of their respective xml files.

Reply:
we need one-time colors at some point to introduce new colors to our theme, which cannot be defined as a variation of existing ones. Think about accent color (`secondaryColor`), for example. 

The way I want to structure the theming is that in all widget styles & xml files under `res/drawable`, we should ONLY reference colors by theme attributes. Ideally we should be able to "paint" every widget with our existing theme colors (slight modifications like alpha is of course possible). 

For instance, a button style references  `?attr/secondaryColor` as its background color. It actually doesn't know that it's blue, it just wants to use the secondaryColor. This will be of great help to us when setting up dark theme. 

If we go with this logic, the color "blue" is only needed once, for the relation `secondaryColor = blue`, but it's actually seen in most places.

---
My question is:
should there still be a color palette somewhere in `colors.xml`? This has the advantage that it lets you see all the colors at once in one place. Keeps it tidy. But it has the disadvantage that, once I create a` @color/blue` resource, it can be referenced from everywhere in the app.

with color palette:
res/colors.xml:
```
<color name="white">#ffffff</color>
<color name="blue">#2492d3</color>
...
```
themes.xml
```
<item name="colorSecondary">@color/blue</item>
```

VS.

without color palette:
themes.xml
```
<item name="colorSecondary">#2492d3</item>
```